src/api/createDeviceMessageHandlingSystem/src/v1/steps.py

```diff
@@ -26,6 +43,64 @@ def parse_mhs_device_payload(data, cache) -> Device:
     return CreateMhsDeviceIncomingParams(**payload)
 
 
+def get_party_key(data, cache) -> str:
+    product: CpmProduct = data[read_product]
+    party_keys = [
+        key.key_value
+        for key in product.keys
+        if key.key_type is ProductKeyType.PARTY_KEY
+    ]
+    try:
+        (party_key,) = party_keys
+    except ValueError:
+        raise NotEprProductError(
+            "Not an EPR Product: Cannot create MHS device for product without exactly one Party Key"
+        )
+    return party_key
+
+
+def check_for_existing_mhs(data, cache):
+    product_team: ProductTeam = data[read_product_team]
+    product: CpmProduct = data[read_product]
+
+    device_repo = DeviceRepository(
+        table_name=cache["DYNAMODB_TABLE"], dynamodb_client=cache["DYNAMODB_CLIENT"]
+    )
+    mhs_device = device_repo.search(
+        product_team_id=product_team.id, product_id=product.id
+    )
+    for device in mhs_device:
+        if device.name == "Product-MHS":
+            raise ConfigurationError(
+                f"There is already an exisitng MHS Device for this product, with device_id: {mhs_device[0].id}"
+            )
```

Comment:
Suggest to 
* fix up the naming
* compress the for/if to one indent
* use a const for MHS device name (will need to be used by ETL)

```suggestion
    devices = device_repo.search(
        product_team_id=product_team.id, product_id=product.id
    )
    if any(device.name == MHS_DEVICE_NAME for device in devices)
        raise ConfigurationError(
            f"There is already an existing MHS Device for this product"
         ) 
```